File: src/ProjectR/content/scripts/spells/FrostBlast.lua
function GetTargetType()
  return Single
end

function GetName()
  return "Frost Blast"
end

function GetDescription()
  return "A chilling blast of frost."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .88
end

function GetDelay()
  return .25
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { ICE }
end

function SpellEffect()
  local damage = (6.25 * aMD * (aICE/100) - 1.25 * dMR) * (100/dICE)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/FrostBlast.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class FrostBlast : SpellScriptBase
    {
        public override string Name { get { return "Frost Blast"; } }
        public override string Description { get { return "A chilling blast of frost."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ICE, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 88; } }
        public override double Delay { get { return .25; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (6.25 * cMD * (cICE / 100) - 1.25 * tMR) * (100 / tICE);
            DealDamage(damage);
        }
    }
}